### PR TITLE
[INJIWEB-1435] - Add fallback logic to use Credential issuer host as authorization server

### PR DESCRIPTION
**Pull request by kongadurgesh**
Align OpenID4VCI issuer metadata with draft-style behavior 
Resolve as candidates from authorization_servers or credential_issuer 
Update tests and issuer config doc

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved robustness of issuer configuration retrieval by handling cases where authorization server information is unavailable in the well-known response.

* **Tests**
  * Added test coverage for issuer configuration retrieval edge cases.

### CODE REVIEW

File: src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java [hunk 1]
```diff
@@ -93,17 +98,64 @@ public IssuersDTO getAllIssuers() throws ApiNotAccessibleException, IOException
...
+     * Per OpenID4VCI, {@code authorization_servers} is optional; when missing or empty, the credential issuer
+     * identifier is used as the authorization server base URL.
+     */
+    private List<String> resolveAuthorizationServerCandidates(CredentialIssuerWellKnownResponse credentialIssuerWellKnownResponse) {
+        List<String> servers = credentialIssuerWellKnownResponse.getAuthorizationServers();
+        if (servers == null || servers.isEmpty()) {
+            return Collections.singletonList(credentialIssuerWellKnownResponse.getCredentialIssuer());
+        }
+        return servers;
+    }
```
[the remark this answers is not in the document]
Comment (kongadurgesh) [reply src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java]: resolved. Please review

**Review by jackjain — COMMENTED** [review]
We should not have a logic to validate grant_types_supported in the issuers endpoint. Lets discuss it once

**Review by jackjain — COMMENTED** [review]
We don't need the additional logic to handle the optional grant type as well as grant type matching for the auth server. It will be done eventually in future as a separate implementation.

File: src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java [hunk 2]
```diff
@@ -27,6 +27,8 @@
 @Validated
 public class IssuersServiceImpl implements IssuersService {
 
+    private static final List<String> DEFAULT_GRANT_TYPES_FOR_COMPATIBILITY = List.of("authorization_code", "implicit");
```
Comment (jackjain) [inline src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java]: Lets remove the default handling for the auth server grant type. Lets pick the first one or send the credential issuer host as is.

**Review by jackjain — APPROVED** [review]
